### DISCUSSION TITLE
how to split ceedling test code in multiple test files linked together at build time

### DISCUSSION THREAD

**Discussion by silabs-theophilel** [ThrowTheSwitch/Ceedling#972] [General]
I have the following

test/test_foo.c
test/test_fii.c
test/test_faa.c

test/utility.c
test/utility.h
All those tests have some code in common that I have refactored into a utility.c file.

I created header files that I included in the individual test files.

The problem is that when I run the tests the symbols are undefined, the linker does not link the two object togethers.

Here is my yml conf

:paths:
  :test:
    - +:test/**
    - -:test/support
  :source:
    - ../../src/**
  :include:
    .. my dependency for mocking
  :support:
    - test/support
  :libraries: []
What special configuration my yaml needs to have to fix this problem?

I tried modyfying the yaml without sucess

**Comment by mvandervoord** [comment]
Hi. 

The problem here is that you haven't told Ceedling that you need it to include `utilities.c` and that you've told it that it CANNOT look for it in the `test` folder (where you currently have it). 

If you've created a standard set of test support functions, by convention, most people put these in the `test/support` folder. You might want to move your files there for clarity. If not, you'll want to add your path to the support path list. This tells Ceedling WHERE to find the support file.

Once there, you simply can `#include` the file in each test, as you would and other C file. The better solution here is if there is a corresponding `utilities.h` file (which you have). It will also work if you directly tell it to include `utilities.c` either with `#include` or `TEST_SOURCE_FILE` with Ceedling 1.0+. Any of these methods tell Ceedling that you want it to build and link the support file.

Between the two tweaks above, it should work as intended. :)

Best of luck!
